UI Enhancements

martyrion opened the pull request:
Please **forgive me** in advance. I made a mess with a lot of unnecessary commits and reverts.

Changes:

- og.css : refactored for reusable styles and codes
- Layer Switcher Control: added drag&drop for ZIndex change. Made sure layers added later via js still update correctly, changes in styling. Dialog closes when you click outside
- GeoImageDragger: changed cursor at corners, made ON/OFF button
- Ruler: made on off button

martyrion committed:
Update og.css

martyrion committed:
Merge branch 'master' of https://github.com/martyrion/openglobus

martyrion committed:
Enhanced GeoImageDragger and mino fix layerSwitcher

martyrion committed:
Delete carrot.png

martyrion committed:
Update billboard.js

martyrion committed:
Changes

martyrion committed:
Merge branch 'master' of https://github.com/martyrion/openglobus

martyrion committed:
UI Improvements

martyrion committed:
UI Updates

martyrion committed:
UI Improvements

- GeoImageDragging : created button, changed cursor on corner
- Ruler : created button
- New UI example .html
- LayerSwitcher: added drag & drop, changing zIndex
- Og.css : refactored styles to be more modular, added global classes for buttons and dialogs

martyrion committed:
Revert "Update billboard.js"

This reverts commit 5ab406fda3b58198eb024cbb85ff69f79e658368.

martyrion committed:
Reverte geoImageDragging.html to original

Zemledelec commented on css/og.css:
```diff
@@ -129,93 +128,170 @@
+/*         MENU V-BAR STYLING      */
+/* =============================== */
+
+.menu-bar-vertical{
```
Please, add prefix **og-**: og-menu..

martyrion replied:
Done

Zemledelec commented on examples/userInterface/userInteface.html:
```diff
@@ -0,0 +1,164 @@
+
+            var sat2 = new XYZ("MapQuest Sat Labels", {
+                fading: true,
+                shininess: 20,
```
Btw, shininess, specular, diffuse, and ambient parameters do not exist anymore. It was a long time ago.

martyrion replied:
Done.

Zemledelec commented on src/og/control/LayerSwitcher.js:
```diff
@@ -5,20 +5,24 @@
         this.dialog = null;
-        this.baseLayersDiv = null;
-        this.overlaysDiv = null;
+        this.baseLayersContainer = null;
```
Maybe for future things, I suggest using prefix $ for dom variables, like this.$baseLayerContainer or even $layers. It's not needed to change now, but if you find it good, we can use this rule.

martyrion replied:
Maybe _**this.DOM_baselayersContainer**_ would be better?

Zemledelec commented on src/og/control/UIhelpers.js:
```diff
@@ -0,0 +1,77 @@
+'use strict';
```
This is cool!:)

Zemledelec commented on src/og/control/ruler/Ruler.js:
```diff
@@ -38,6 +45,26 @@ class Ruler extends Control {
+
+
+    createMenuBtn() {
+        let btn = elementFactory('div', { id: 'ruler-menu-btn', class: 'ruler menu-btn OFF' },
```
need prefix **og-**

martyrion replied:
Done. you want the prefix og- to be present in all css classes and ids? In my opinion it reduces a bit the readability of the code.

I mean we are in the src/ of Openglobus after all

Zemledelec commented on src/og/control/ruler/RulerScene.js:
```diff
@@ -112,12 +112,14 @@ class RulerScene extends RenderNode {
             polygonOffsetUnits: -1.0,
-            relativeToGround: true
+            relativeToGround: true,
+            displayInLayerSwitcher: false
```
Nice!

Zemledelec replied:
Also `displayInLayerSwitcher` must be in the `Layer `class constructor.

martyrion replied:
I think it is already. I added it to the  @params of Layer.js

Zemledelec reviewed (changes requested):
Super! Just a few improves needed, I've made comments!
FYI @pavletto if you have something please respond.

martyrion committed:
Removed deprecated shininess, specular, diffuse, ambient

martyrion committed:
Changed menu-bar-vertical to og-menu....

martyrion committed:
Added og- prefix to Ruler.js and .css

martyrion committed:
Fixed a minor issue caused by latest css class rename

martyrion committed:
Update Layer.js @params with displayInLayerSwitcher

Zemledelec commented on css/og.css:
```diff
@@ -129,93 +128,170 @@
+/* =============================== */
+
+/* All buttons have this class + OFF if they are OFF */
+.menu-btn{
```
Sorry, I had to say that adding **og-** touches every style name we use in native controls. Please, add **og-** for all new styles.

Zemledelec commented on css/og.css:
```diff
@@ -129,93 +128,170 @@
+#layer-switcher-menu-btn > div{
+    background: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAYAAABw4pVUAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAlPSURBVHgB7Z39edNIEMbf3HP/k6uAoQJ8FSAqIB1YVECugiwVECqwrgKgApsKAhVIVECugpwnlogiz0izH5Ll4N/zLB8r7Wq1r2Z2VlpLZ5g/59uUbdPzbXpR/011fpPa3Napqv/+sU3lNn3fpm91ni/rug0am216jScKbdO7bVph15F3idPNNl3AD0u9T4psm64xjgDdVMCf30IQdjVX2F2xdxOlQmnHECkEsRznICyws4af ...
 
+.layer-switcher.dialog { 
```
og-

Zemledelec commented on css/og.css:
```diff
@@ -129,93 +128,170 @@
-   display: flex;
-   align-items: center;
-   height: 24px;
+.layer-container{
```
og- etc.

Zemledelec commented on css/og.css:
```diff
@@ -129,93 +128,170 @@
+
+/* Just set position and background url */
+
+#geo-image-dragger-menu-btn{
```
og-

Zemledelec reviewed (changes requested):
Forget to mention that **og-** for all styles for native controls needed.

martyrion committed:
Added Terrain Switcher

terrain option of globe, can be an array. First item will be rendered. Still works if only one object is passed to terrain option.

Added _terrainPool option to Planet.js. This is just to keep the terrain options.

martyrion committed:
Og-ged everything! :)

martyrion committed:
Layer record contents alignment

Zemledelec reviewed (approved):
Good!
